x/gamm/pool-models/balancer/pool.go

```diff
@@ -19,7 +19,7 @@ import (
 const (
 	nonPostiveSharesAmountErrFormat = "shares amount must be positive, was %s"
 	nonPostiveTokenAmountErrFormat  = "token amount must be positive, was %s"
-	sharesLargerThanMaxErrFormat    = "%s resulted shares is larger than the max amount of %s"
+	sharesLargerThanMaxErrFormat    = "resulted shares %s should be lower than the max amount of %s"
```

Comment:
I don't think this is solving the problem presented in the issue. The problem seems that incorrect values are being formatted

Reply:
So we could just subtract one from the max amount, how it was proposed [here](https://github.com/osmosis-labs/osmosis/issues/3427#issuecomment-1319369089)?